crates/trippy-core/src/tos.rs

```diff
@@ -0,0 +1,206 @@
+use crate::{Error, TypeOfService};
+use std::fmt::Display;
+
+/// Explicit Congestion Notification (`ECN`).
+///
+/// This is used in the `ECN` field of the `IP` header.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum ExplicitCongestionNotification {
+    /// Not ECN-Capable Transport (00, 0 dec)
+    NotECT,
+    /// ECN Capable Transport(1) (01, 1 dec)
+    ECT1,
+    /// ECN Capable Transport(0) (10, 2 dec)
+    ECT0,
+    /// Congestion Experienced (11, 3 dec)
+    CE,
+}
+
+impl TryFrom<u8> for ExplicitCongestionNotification {
+    type Error = Error;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(Self::NotECT),
+            1 => Ok(Self::ECT1),
+            2 => Ok(Self::ECT0),
+            3 => Ok(Self::CE),
+            _ => Err(Error::Other(format!("invalid ECN value: {value}"))),
+        }
+    }
+}
+
+impl Display for ExplicitCongestionNotification {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::NotECT => write!(f, "Not ECN-Capable Transport"),
+            Self::ECT1 => write!(f, "ECN Capable Transport(1)"),
+            Self::ECT0 => write!(f, "ECN Capable Transport(0)"),
+            Self::CE => write!(f, "Congestion Experienced"),
+        }
+    }
+}
+
+/// Differentiated Services Code Point (`DSCP`).
+///
+/// This is used in the `DSCP` field of the `IP` header.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum DifferentiatedServicesCodePoint {
+    /// Best Effort (000 000, 0 dec)
+    BE,
+    /// Assured Forwarding 11 (001 010, 10 dec)
+    AF11,
+    /// Assured Forwarding 12 (001 100, 12 dec)
+    AF12,
+    /// Assured Forwarding 13 (001 110, 14 dec)
+    AF13,
+    /// Assured Forwarding 21 (010 010, 18 dec)
+    AF21,
+    /// Assured Forwarding 22 (010 100, 20 dec)
+    AF22,
+    /// Assured Forwarding 23 (010 110, 22 dec)
+    AF23,
+    /// Assured Forwarding 31 (011 010, 26 dec)
+    AF31,
+    /// Assured Forwarding 32 (011 100, 28 dec)
+    AF32,
+    /// Assured Forwarding 33 (011 110, 30 dec)
+    AF33,
+    /// Assured Forwarding 41 (100 010, 34 dec)
+    AF41,
+    /// Assured Forwarding 42 (100 100, 36 dec)
+    AF42,
+    /// Assured Forwarding 43 (100 110, 38 dec)
+    AF43,
+    /// Class Selector 1 (001 000, 8 dec)
+    CS1,
+    /// Class Selector 2 (010 000, 16 dec)
+    CS2,
+    /// Class Selector 3 (011 000, 24 dec)
+    CS3,
+    /// Class Selector 4 (100 000, 32 dec)
+    CS4,
+    /// Class Selector 5 (101 000, 40 dec)
+    CS5,
+    /// Class Selector 6 (110 000, 48 dec)
+    CS6,
+    /// Class Selector 7 (111 000, 56 dec)
+    CS7,
+    /// High Priority Expedited Forwarding (101 110, 46 dec)
+    EF,
+    /// Other DSCP value (not defined in the standard)
+    Other(u8),
```

Comment:
I think it is worth adding a few more named PHBs/DSCPs that occasionally crop up in the wild:
VoiceAdmit/VA (rfc5865) decimal 44 (101 100)
LowerEffort/LE (rfc8622) decimal 1 (000 001)

Reply:
Added